engine/src/flutter/lib/web_ui/lib/src/engine/semantics/landmarks.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'label_and_value.dart';
+import 'semantics.dart';
+
+/// Indicates a complementary element.
+///
+/// Uses aria complementary role to convey this semantic information to the element.
+///
+/// Screen-readers takes advantage of "aria-label" to describe the visual.
+class SemanticComplementary extends SemanticRole {
+  SemanticComplementary(SemanticsObject semanticsObject)
+    : super.withBasics(
+        EngineSemanticsRole.complementary,
+        semanticsObject,
+        preferredLabelRepresentation: LabelRepresentation.ariaLabel,
+      ) {
+    setAriaRole('complementary');
+  }
+
+  @override
+  void update() {
+    super.update();
+    if (semanticsObject.label != null) {
+      setAttribute('aria-label', semanticsObject.label!);
```

Comment:
you don't need this if you use `withBasics` constructor, they should manage the label correctly.